addons/mass_mailing_sale/models/mailing_mailing.py

```diff
@@ -10,52 +10,59 @@ class MassMailing(models.Model):
 
     @api.depends('mailing_domain')
     def _compute_sale_quotation_count(self):
-        has_so_access = self.env['sale.order'].check_access_rights('read', raise_exception=False)
-        if not has_so_access:
-            self.sale_quotation_count = 0
-            return
         for mass_mailing in self:
-            mass_mailing.sale_quotation_count = self.env['sale.order'].search_count(mass_mailing._get_sale_utm_domain())
+            mass_mailing.sale_quotation_count = self.env['sale.order'].sudo().search_count(
+                [('source_id', '=', mass_mailing.source_id.id)]
+            )
 
     @api.depends('mailing_domain')
     def _compute_sale_invoiced_amount(self):
-        if not self.user_has_groups('sales_team.group_sale_salesman') or not self.user_has_groups('account.group_account_invoice'):
-            self.sale_invoiced_amount = 0
-            return
         for mass_mailing in self:
             domain = expression.AND([
-                mass_mailing._get_sale_utm_domain(),
+                [('source_id', '=', mass_mailing.source_id.id)],
                 [('state', 'not in', ['draft', 'cancel'])]
             ])
-            moves = self.env['account.move'].search_read(domain, ['amount_untaxed_signed'])
+            moves = self.env['account.move'].sudo().search_read(domain, ['amount_untaxed_signed'])
             mass_mailing.sale_invoiced_amount = sum(i['amount_untaxed_signed'] for i in moves)
 
     def action_redirect_to_quotations(self):
-        action = self.env["ir.actions.actions"]._for_xml_id("sale.action_quotations_with_onboarding")
-        action['domain'] = self._get_sale_utm_domain()
-        action['context'] = {'create': False}
-        return action
+        helper_header = _("No Quotations yet!")
+        helper_message = _("Quotations will appear here once your customers add "
+                           "products to their Carts or when your sales reps assign this mailing.")
+        return {
+            'context': {'create': False, 'search_default_group_by_day': True},
+            'domain': [('source_id', '=', self.source_id.id)],
+            'help': f'<p class="o_view_nocontent_smiling_face">{helper_header}</p><p>{helper_message}</p>',
+            'name': _("Sales Analysis"),
+            'res_model': 'sale.report',
+            'search_view_id': [self.env.ref('mass_mailing_sale.sale_report_view_search').id, 'search'],
+            'type': 'ir.actions.act_window',
+            'view_mode': 'graph,pivot,tree,form',
+        }
 
     def action_redirect_to_invoiced(self):
-        action = self.env["ir.actions.actions"]._for_xml_id("account.action_move_out_invoice_type")
-        moves = self.env['account.move'].search(self._get_sale_utm_domain())
-        action['context'] = {
-            'create': False,
-            'edit': False,
-            'view_no_maturity': True
+        moves = self.env['account.move'].search([('source_id', '=', self.source_id.id)])
```

Comment:
Why not incorporate this condition in the domain?

Reply:
There are two domains because we need to look in two different models. I still changed this part a bit because it wasn't working as intended.